resolver/src/hosts.rs

```diff
@@ -0,0 +1,121 @@
+use std::collections::HashMap;
+use std::fs;
+use std::time::{Duration, SystemTime};
+use std::io::{BufRead, BufReader};
+use std::fs::File;
+use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
+use std::sync::Mutex;
+
+lazy_static! {
+    static ref CACHE_MAX_AGE: Duration = Duration::new(5, 0);
+    static ref HOSTS: Mutex<Hosts> = Mutex::new(Hosts::new());
+}
+
+#[derive(Debug)]
+pub struct Hosts {
+    by_name: HashMap<String, Vec<IpAddr>>,
+    expire: SystemTime,
+    path: String,
+    mtime: SystemTime,
+    size: u64,
+}
+
+impl Hosts {
```

Comment:
It would be good to have tests over various file contents. This can probably just be buffers of data and unit tests in this file.